icann-rdap-client/src/md/redacted.rs

```diff
@@ -61,3 +68,501 @@ impl ToMd for &[Redacted] {
+    pub replacement_path: Option<String>,
+    pub method: Value,                         // the method they are using
+    pub reason: Value,                         // the reason
+    pub result_type: Vec<Option<ResultType>>,  // a vec of our own internal Results we found
+    pub redaction_type: Option<RedactionType>, //
+}
+
+// This isn't just based on the string type that is in the redaction method, but also based on the result type above
+#[derive(Debug, PartialEq, Clone)]
+pub enum RedactionType {
+    EmptyValue,
+    PartialValue,
+    ReplacementValue,
+    Removal,
+    Unknown,
+}
+
+// this is our public entry point
+pub fn replace_redacted_items(orignal_response: RdapResponse) -> RdapResponse {
+    // convert the RdapResponse to a string
+    let rdap_json = serde_json::to_string(&orignal_response).unwrap();
+    // convert the string to a JSON Value
+    let mut rdap_json_response: Value = serde_json::from_str(&rdap_json).unwrap();
+    // Initialize the final response with the original response
+    let mut response = orignal_response;
+    // pull the redacted array out of the JSON
+    let redacted_array_option = rdap_json_response["redacted"].as_array().cloned();
+
+    // if there are any redactions we need to do some modifications
+    if let Some(ref redacted_array) = redacted_array_option {
+        parse_redacted_json(&mut rdap_json_response, Some(redacted_array));
+        // convert the Value back to a RdapResponse
+        response = serde_json::from_value(rdap_json_response).unwrap();
+    } // END if there are redactions
+
+    // send the response back so we can display it to the client
+    response
+}
+
+fn parse_redacted_json(
+    rdap_json_response: &mut serde_json::Value,
+    redacted_array_option: Option<&Vec<serde_json::Value>>,
+) {
+    if let Some(redacted_array) = redacted_array_option {
+        let redactions = parse_redacted_array(rdap_json_response, redacted_array);
+        // Loop through the RedactedObjects
+        for redacted_object in redactions {
+            // If we have determined we are doing some kind of substitution
+            if redacted_object.do_final_path_subsitution && !redacted_object.final_path.is_empty() {
+                let path_count = redacted_object.path_index_count as usize;
+                for path_index_count in 0..path_count {
+                    let final_path_option = &redacted_object.final_path[path_index_count];
+                    if let Some(final_path) = final_path_option {
+                        // This is a replacement and we SHOULD NOT be doing this until it is sorted out.
+                        // For experimental reasons though, we shall continue.
+                        if let Some(redaction_type) = &redacted_object.redaction_type {
+                            if *redaction_type == RedactionType::ReplacementValue {
+                                // let replacement_path_str;
```

Comment:
Why is all this commented code here?

Reply:
We can rip that out, I left it in for the fleeting chance that we change out mind and decided to do the copy version of replacementValue. After todays convo I can rip it out.